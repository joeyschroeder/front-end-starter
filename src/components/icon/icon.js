import React from 'react';

import PropTypes from 'prop-types';
import classNames from 'classnames';

export function Icon(props) {
  const { className: classNameProp, name } = props;
  const className = classNames('fa', 'fa-fw', classNameProp, {
    [`fa-${name}`]: name,
  });

  return <i aria-hidden="true" className={className} />;
}

Icon.propTypes = {
  className: PropTypes.string,
  name: PropTypes.string,
};

Icon.defaultProps = {
  className: '',
  name: 'bell-o',
};
