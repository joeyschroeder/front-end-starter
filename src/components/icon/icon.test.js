import { Icon } from './icon';
import React from 'react';
import { create } from 'react-test-renderer';

describe('<Icon />', () => {
  it('should render correctly', () => {
    const component = create(<Icon />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
