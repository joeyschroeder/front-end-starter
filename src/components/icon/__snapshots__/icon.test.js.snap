// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should render correctly 1`] = `
<i
  aria-hidden="true"
  className="fa fa-fw fa-bell-o"
/>
`;
